assets/server/static/js/popup/main.js

```diff
@@ -12,28 +12,35 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-window.addEventListener("DOMContentLoaded", async () => {
-  const originElement = document.getElementById("origin");
-  const windowElement = document.getElementById("windowname");
+const originElement = document.querySelector("#origin");
+const windowElement = document.querySelector("#windowname");
 
-  // leverage the URL parameter provided from the request and set it to the target origin
-  const encodedUriComponent = new URLSearchParams(window.location.search).get("origin");
-  if (!encodedUriComponent && !originElement) {
-    alert("An origin URL parameter must be provided.")
+if (!originElement) {
+  alert("The origin input was not detected.");
+  window.close();
```

Comment:
I think we need a `return` after all of these to prevent execution from continuing.

Reply:
What error?

Reply:
by removing the method scope i was triggering the error, bringing the returns back.

Reply:
I didn't follow this.

Reply:
You caused the error when you removed the addEventListener but if you add that back the returns will work?

Reply:
yes, i think its because the the second argument to addEventListener is a method. So you can return inside that method. Without the addEventListener, return triggers an error: 

```
Uncaught SyntaxError: Illegal return statement
```

Reply:
Right, adding that back with the return should be good.